tensorflow/compiler/mlir/xla/transforms/hlo_legalize_to_lhlo.cc

```diff
@@ -409,37 +381,31 @@ class StdToLhloReturnOpConverter : public OpConversionPattern<mlir::ReturnOp> {
     auto funcOp = returnOp.getParentOfType<FuncOp>();
     auto numFuncArgs = funcOp.getNumArguments();
     auto loc = returnOp.getLoc();
-
+    auto block = returnOp.getOperation()->getBlock();
     for (auto operand : llvm::enumerate(operands)) {
       auto returnArgNumber = numFuncArgs - numReturnValues + operand.index();
       auto dstBuffer = funcOp.getArgument(returnArgNumber);
       if (dstBuffer == operand.value()) {
         continue;
       }
 
-      auto dealloc = FindInsertionPointForCopy(operand.value());
-
-      if (dealloc == nullptr) {
-        returnOp.emitOpError()
-            << "Missing dealloc for operand " << operand.index();
-        return matchFailure();
-      }
-      OpBuilder::InsertionGuard guard(rewriter);
-      rewriter.setInsertionPoint(dealloc);
+      rewriter.setInsertionPointToEnd(block);
```

Comment:
Why is this needed?

Reply:
Resolved. This conversion pattern is going to be moved to [Buffer Assignment ](https://github.com/tensorflow/tensorflow/pull/37212) and changed to a template pattern converter.

```c++
template <typename ReturnOpSourceTy, typename ReturnOpTargetTy,
          typename CopyOpTy>
class NonVoidToVoidReturnOpConversion
    : public OpConversionPattern<ReturnOpSourceTy> {
```